src/pubsub/ua_pubsub_writer.c

```diff
@@ -2131,23 +2171,26 @@ generateNetworkMessage(UA_PubSubConnection *connection, UA_WriterGroup *wg,
 
 static UA_StatusCode
 sendBufferedNetworkMessage(UA_Server *server, UA_PubSubConnection *connection,
-                           UA_NetworkMessageOffsetBuffer *buffer,
+                           UA_ByteString *buffer,
                            UA_ExtensionObject *transportSettings) {
-    if(UA_NetworkMessage_updateBufferedMessage(buffer) != UA_STATUSCODE_GOOD)
-        UA_LOG_DEBUG(&server->config.logger, UA_LOGCATEGORY_SERVER,
-                     "PubSub sending. Unknown field type.");
-    return connection->channel->send(connection->channel,
-                                     transportSettings, &buffer->buffer);
+
+    return connection->channel->send(connection->channel, transportSettings, buffer);
 }
 
 #ifdef UA_ENABLE_PUBSUB_ENCRYPTION
 static UA_StatusCode
 encryptAndSign(UA_WriterGroup *wg, const UA_NetworkMessage *nm,
-                 UA_Byte *signStart, UA_Byte *encryptStart,
-                 UA_Byte *msgEnd) {
+               UA_Byte *signStart, UA_Byte *encryptStart,
+               UA_Byte *msgEnd) {
     UA_StatusCode rv;
     void *channelContext = wg->securityPolicyContext;
 
+    size_t sigSize = wg->config.securityPolicy->symmetricModule.cryptoModule.
+                     signatureAlgorithm.getLocalSignatureSize(channelContext);
+
+    if(wg->config.rtLevel == UA_PUBSUB_RT_FIXED_SIZE)
```

Comment:
I don't understand why this is required.
This should be documented in a good comment.
I prefer to have uniform code paths for rt/non-rt as much as possible.